teamscale-gradle-plugin/src/main/kotlin/com/teamscale/TeamscalePlugin.kt

```diff
@@ -73,44 +79,90 @@ open class TeamscalePlugin : Plugin<Project> {
-            configureTestImpactedTask(
-                project,
-                pluginExtension,
-                testImpactedTask
-            )
-        }
+        // Add the teamscale extension also to all TestImpacted tasks
+        extendTestImpactedTasks(project, pluginExtension)
+        extendTestTasks(project)
+        extendJaCoCoReportTasks(project)
     }
 
-
-    /** Configures the given impacted test executor. */
-    private fun configureTestImpactedTask(
+    private fun extendTestImpactedTasks(
         project: Project,
-        pluginExtension: TeamscalePluginExtension,
-        testImpacted: TestImpacted
+        pluginExtension: TeamscalePluginExtension
     ) {
-        project.logger.info("Configuring impacted tests executor task for ${project.name}:${testImpacted.name}")
-
-        val extension = pluginExtension.applyTo(testImpacted)
-
-        testImpacted.apply {
-            taskExtension = extension
-            dependsOn.add(project.configurations.getByName(impactedTestEngineConfiguration))
+        project.tasks.withType(TestImpacted::class.java) { testImpactedTask ->
+            /** Configures the given impacted test executor. */
+            project.logger.info(
+                "Configuring impacted tests executor task for ${project.name}:${
+                    testImpactedTask
+                        .name
+                }"
+            )
+            val extension = pluginExtension.applyTo(testImpactedTask)
+            testImpactedTask.apply {
+                taskExtension = extension
+                this.pluginExtension = pluginExtension
+                dependsOn.add(project.configurations.getByName(impactedTestEngineConfiguration))
+            }
+            val teamscaleReportTask = project.rootProject.tasks
+                .maybeCreate("${testImpactedTask.name}Report", TestwiseCoverageReportTask::class.java)
+            testImpactedTask.finalizedBy(teamscaleReportTask)
+            testImpactedTask.reportTask = teamscaleReportTask
+            teamscaleReportTask.apply {
+                testTaskName = testImpactedTask.name
+                configuration = extension
+            }
+            teamscaleReportTask.uploadTask = teamscaleUploadTask
+            registerReportAfterTask(testImpactedTask, extension.report)
         }
+    }
 
-        val teamscaleReportTask = project.rootProject.tasks
-            .maybeCreate("${testImpacted.name}Report", TeamscaleReportTask::class.java)
-        testImpacted.finalizedBy(teamscaleReportTask)
-
-        testImpacted.reportTask = teamscaleReportTask
-
-        teamscaleReportTask.apply {
-            testTaskName = testImpacted.name
-            configuration = extension
+    /**
+     * Adds an extension to the JacocoReport tasks and adds the report
+     * to the upload task when the extension has been configured.
+     */
+    private fun extendJaCoCoReportTasks(project: Project) {
```

Comment:
AFAIS in the other identifiers you write Jacoco (i.e. no capitals for CoCo ;-) unify?